legacy/src/main/java/org/opensearch/sql/legacy/pit/PointInTimeHandlerImpl.java

```diff
@@ -37,37 +39,68 @@ public PointInTimeHandlerImpl(Client client, String[] indices) {
     this.indices = indices;
   }
 
-  /** Create PIT for given indices */
+  /**
+   * Constructor for class
+   *
+   * @param client OpenSearch client
+   * @param pitId Point In Time ID
+   */
+  public PointInTimeHandlerImpl(Client client, String pitId) {
+    this.client = client;
+    this.pitId = pitId;
+  }
+
+  /**
+   * Create PIT for given indices
+   *
+   * @return Point In Time creation status
+   */
   @Override
-  public void create() {
+  public boolean create() {
     CreatePitRequest createPitRequest =
         new CreatePitRequest(
             LocalClusterState.state().getSettingValue(SQL_CURSOR_KEEP_ALIVE), false, indices);
     client.createPit(
         createPitRequest,
         new ActionListener<>() {
+
           @Override
           public void onResponse(CreatePitResponse createPitResponse) {
             pitId = createPitResponse.getId();
+            createStatus = true;
             LOG.info("Created Point In Time {} successfully.", pitId);
           }
 
           @Override
           public void onFailure(Exception e) {
+            createStatus = false;
             LOG.error("Error occurred while creating PIT", e);
           }
         });
+    while (createStatus == null) {
+      try {
+        Thread.sleep(100);
+      } catch (InterruptedException e) {
+        LOG.error("Error occurred while creating PIT", e);
+      }
+    }
+    return createStatus;
```

Comment:
What is this? This is bad. Are there no synchronous requests?

Reply:
Looking at the previous checked in iteration of this change here https://github.com/opensearch-project/sql/pull/2703/files#diff-b7d29ea36ecb24676a833f1897d949d465f34c1595dbdd81c283a355447a9704R42 I see this change is made after the previous PR and I do not have entire context on why would we need this change. Looks like this might be part of some testing. I took out this unnecessary change from this PR and verified the changes.

Reply:
I added completable future to wait for the async thread to complete PIT creation/delete before the process continues with next operation which might be depending on the PIT ID. This way its safe to assume we are not failing during query execution if PIT is still being created.